compiler/rustc_target/src/spec/targets/xtensa_esp32_none_elf.rs

```diff
@@ -0,0 +1,24 @@
+use crate::spec::{base::xtensa, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        llvm_target: "xtensa-none-elf".into(),
+        pointer_width: 32,
+        data_layout: "e-m:e-p:32:32-v1:8:8-i64:64-i128:128-n32".into(),
```

Comment:
This seem to be wrong? I got the following error when building for this target:

```
  error: data-layout for target `xtensa-esp32-none-elf`, `e-m:e-p:32:32-v1:8:8-i64:64-i128:128-n32`, differs from LLVM target's `xtensa-none-elf` default layout, `e-m:e-p:32:32-i8:8:32-i16:16:32-i64:64-n32`
```

https://github.com/taiki-e/portable-atomic/actions/runs/9492602960/job/26160155007#step:6:1583

Reply:
Rust is not currently using a new enough LLVM to include the patches to generate code (discussed here: https://github.com/rust-lang/rust/pull/125141#issuecomment-2137459934), so for now, you still need the esp toolchain in your CI. I think by switching to LLVM 18, or latest LLVM 19 we should be able to build _some_ basic binaries.